CMakeLists.txt

```diff
@@ -3,52 +3,59 @@ cmake_minimum_required(VERSION 3.14.1)
 project(rpcs3)
 
 if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
-	if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
-		message(FATAL_ERROR "RPCS3 requires at least gcc-9.")
-	endif()
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
+        message(FATAL_ERROR "RPCS3 requires at least gcc-9.")
+    endif()
 elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
-	if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11.0)
-		message(FATAL_ERROR "RPCS3 requires at least clang-11.0.")
-	endif()
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11.0)
+        message(FATAL_ERROR "RPCS3 requires at least clang-11.0.")
+    endif()
 endif()
 
 # include build options
 include(options.cmake)
 
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/rpcs3/cmake_modules")
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake_modules")
 
 set(CMAKE_CXX_STANDARD 20)
 include(CheckCXXCompilerFlag)
 
-if(NOT CMAKE_BUILD_TYPE)
-	message(STATUS "No build type selected, default to Release")
-	set(CMAKE_BUILD_TYPE "Release")
+# get build type
```

Comment:
Yeah this chunk of code is unnecessary clutter.
You don't need to warn the user that debug builds are unstable, if they've decided to compile RPCS3 with debug options we can assume they know what they're doing. In addition, you don't handle RelWithDebInfo.
Best to just leave it as it was.

Reply:
removed it from the commit, but i do still think its messy and can lead to undefined behavior. ill revisit it again some other time :)